src/java.base/share/classes/java/util/PrimitiveIterator.java

```diff
@@ -68,10 +68,18 @@
 public interface PrimitiveIterator<T, T_CONS> extends Iterator<T> {
 
     /**
-     * Performs the given action for each remaining element, in the order
-     * elements occur when iterating, until all elements have been processed
-     * or the action throws an exception.  Errors or runtime exceptions
-     * thrown by the action are relayed to the caller.
+     * Performs the given action for each remaining element until all elements
+     * have been processed or the action throws an exception.  Actions are
+     * performed in the order of iteration, if that order is specified.
+     * Exceptions thrown by the action are relayed to the caller.
+     * <p>
+     * The behavior of an iterator is unspecified if the action modifies the
+     * collection in any way (even by calling the {@link #remove remove} method
```

Comment:
It's kind of odd to mention "the collection" here. Iterator is defined as being over a collection, but strictly speaking this isn't true; it can be an iterator over anything. PrimitiveIterator doesn't say anything more specific than this, just that it's a base for iterating primitives... and collections cannot contain primitives.

I'm not sure what a better term for this is. Something like, "the underlying source of the Iterator"?

Reply:
Yeah, well spotted. I agree it's awkward. How about we lean on the behavior of the boxed counterpart:
```java
    /**
     * Performs the given action for each remaining element until all elements
     * have been processed or the action throws an exception.  Actions are
     * performed in the order of iteration, if that order is specified.
     * <p>
     * This primitive-based method conforms to the same behavior as its
     * boxed counterpart with regards to the action's behavior.
?
```

Reply:
Referring to the "boxed counterpart" is a bit too oblique, I think. The specification of Iterator itself isn't all that good to begin with. It's written as if the only possible source of an Iterator is a collection (which might have been true at first, but it isn't true any longer). But the rest of the Iterator spec refers to "the collection" so it kind of makes sense in that context.

Maybe just making a minimal change from "collection" to "source of elements" would make the most sense.

     * The behavior of an iterator is unspecified if the action modifies the source of
     * elements in any way (even by calling the {@link #remove remove} method

Reply:
Ok, that works for me.